src/rgw/rgw_http_client_curl.h

```diff
@@ -0,0 +1,35 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2018 SUSE Linux GmBH
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+#ifndef RGW_HTTP_CLIENT_CURL_H
+#define RGW_HTTP_CLIENT_CURL_H
+
+#include <map>
+#include "rgw_frontend.h"
+
+namespace rgw {
+namespace curl {
+
+using fe_map_t = std::multimap <std::string, RGWFrontendConfig *>;
+
+void init_ssl();
+void setup_curl(const fe_map_t& m);
+void cleanup_curl();
+
+namespace openssl {
+void init_ssl();
+}
```

Comment:
we only try to call this `openssl::init_ssl()` from an `#ifdef WITH_CURL_OPENSSL` block in `rgw_http_client_curl.cc`. so i don't think it needs to be declared here in the header, or defined when `WITH_CURL_OPENSSL` isn't